wpilibj/src/main/java/edu/wpi/first/wpilibj/AnalogInput.java

```diff
@@ -334,9 +332,28 @@ public void setSimDevice(SimDevice device) {
     AnalogJNI.setAnalogInputSimDevice(m_port, device.getNativeHandle());
   }
 
-  @Override
-  public void initSendable(SendableBuilder builder) {
-    builder.setSmartDashboardType("Analog Input");
-    builder.addDoubleProperty("Value", this::getAverageVoltage, null);
+  public static class AnalogInputSendable implements Sendable<AnalogInput> {
+    @Override
+    public Class<AnalogInput> getTypeClass() {
+      return AnalogInput.class;
+    }
+
+    @Override
+    public String getTypeString() {
+      return "Analog Input";
+    }
+
+    @Override
+    public boolean isClosed(AnalogInput obj) {
+      return obj.m_port == 0;
+    }
+
+    @Override
+    public void initSendable(AnalogInput obj, SendableTable table) {
+      table.publishDouble("Value", obj::getAverageVoltage);
+    }
   }
+
+  /** Sendable for serialization. */
+  public static final AnalogInputSendable sendable = new AnalogInputSendable();
```

Comment:
Any reason to prefer a named class with a single static instance over an anonymous class?

Reply:
You can't do a static inner anonymous class.  See https://stackoverflow.com/questions/758570/is-it-possible-to-make-anonymous-inner-classes-in-java-static